compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingOps.cpp

```diff
@@ -106,32 +106,31 @@ EncodingAttr EncodingAttr::get(MLIRContext *ctx, EncodingRole role,
   auto optionalToAttr = [&](std::optional<int64_t> x) {
     return x ? b.getIndexAttr(*x) : IntegerAttr();
   };
-  auto roleAttr = EncodingRoleAttr::get(ctx, role);
   auto origTypeAttr = origType ? TypeAttr::get(origType) : TypeAttr();
   auto roundDimsToAttr = roundDimsTo.empty()
                              ? DenseI64ArrayAttr()
                              : b.getDenseI64ArrayAttr(roundDimsTo);
-  return get(ctx, roleAttr, b.getTypeArrayAttr(elemTypes), origTypeAttr,
+  return get(ctx, IntegerAttr::get(IntegerType::get(ctx, 64), operandIndex),
```

Comment:
nit: I think we can use `b.getI64IntegerAttr(operandIndex)`.

https://github.com/llvm/llvm-project/blob/aaf50bf34f3a2007221c644384d238666cfc2bc3/mlir/include/mlir/IR/Builders.h#L133C15-L133C32

`int64_t` type looks okay to me, although others are using `index` type. Do you know why we decide to use `int64_t` in this case?

Reply:
Indeed there are very subtle differences. Updated to be conformant with the surrounding code.